### ISSUE TITLE
use ObservableArray for Particle collections?

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/gas-properties#116]
To optimize performance, I'm currently using plain old Arrays for storing Particles.  I didn't want to involve ObservableArray because of the overhead.,  

But this led to an order dependency in https://github.com/phetsims/gas-properties/issues/113#issuecomment-500665501 when deriving `numberOfParticlesProperty`.  And I'm guessing that PhET-iO is going to want to get at the collection of particles in the future.  So I'm going to investigate the performance impact of switching to ObservableArray.  

The primary change would be in ParticleSystem.  Instead of this:

```js
this.heavyParticles = []; // {HeavyParticle[]} heavy particles inside the container
this.lightParticles = []; // {LightParticle[]} light particles inside the container
this.heavyParticlesOutside = []; // {HeavyParticle[]} heavy particles outside the container
this.lightParticlesOutside = []; // {LightParticle[]} light particles outside the container
```

we'd have this:

```js
// @public (read-only) {ObservableArray.<HeavyParticle>} heavy particles inside the container
this.heavyParticles = new ObservableArray( [] );

// @public (read-only) {ObservableArray.<LightParticle>} light particles inside the container
this.lightParticles = new ObservableArray( [] );

// @public (read-only) {ObservableArray.<HeavyParticle>} heavy particles outside the container
this.heavyParticlesOutside = new ObservableArray( [] );

// @public (read-only) {ObservableArray.<LightParticle>} light particles outside the container
this.lightParticlesOutside = new ObservableArray( [] );
```

**Comment by pixelzoom** [comment]
I have this working in my local copy.  I feel like I've done everything possible to use ObservableArray efficiently, including iterating over its raw array where possible.  But the performance is noticeably degraded, even on my latest-model MacBook Pro.   I could commit to a branch, but I'm not convinced that there's much point in doing that.  I'll sleep on it...

**Comment by pixelzoom** [comment]
These changes will not go to master.  I've saved them temporarily in observable-array-116, with no plans to merge.

**Comment by pixelzoom** [comment]
For posterity, I published one-off [1.0.0-observableArray116.1](https://phet-dev.colorado.edu/html/gas-properties/1.0.0-observableArray116.1/phet/gas-properties_en_phet.html) from this branch.

**Comment by pixelzoom** [comment]
So... I'm going to stick with plain old Arrays, and punt on ObservableArray.   Closing.
